From 9a0ac6d9c21b914f887d2a1e0051b5beffa81bd7 Mon Sep 17 00:00:00 2001
From: Richard Weinberger <richard@nod.at>
Date: Tue, 9 Jul 2013 11:05:14 +0100
Subject: [PATCH] LXC: Create /dev/tty within a container

Many applications use /dev/tty to read from stdin.
e.g. zypper on openSUSE.

Let's create this device node to unbreak those applications.
As /dev/tty is a synonym for the current controlling terminal
it cannot harm the host or any other containers.

Signed-off-by: Richard Weinberger <richard@nod.at>
Index: libvirt-1.1.0/src/lxc/lxc_container.c
===================================================================
--- libvirt-1.1.0.orig/src/lxc/lxc_container.c
+++ libvirt-1.1.0/src/lxc/lxc_container.c
@@ -839,6 +839,7 @@ static int lxcContainerPopulateDevices(c
         { LXC_DEV_MAJ_MEMORY, LXC_DEV_MIN_FULL, 0666, "/dev/full" },
         { LXC_DEV_MAJ_MEMORY, LXC_DEV_MIN_RANDOM, 0666, "/dev/random" },
         { LXC_DEV_MAJ_MEMORY, LXC_DEV_MIN_URANDOM, 0666, "/dev/urandom" },
+        { LXC_DEV_MAJ_TTY, LXC_DEV_MIN_TTY, 0666, "/dev/tty" },
     };
     const struct {
         const char *src;
